File: tesla_proj/backend/classes/track.py
class Track:
    def __init__ (self, track_data):
        self.set_track_data(track_data)
        
    def set_track_data(self, track_data):
        self.source = track_data["uri"]
        self.name = track_data["name"]
        self.artist = track_data["album"]["artists"][0]["name"]
        self.album = track_data["album"]["name"]
        try:
            self.album_cover = track_data["album"]["images"][1]["url"]
        except IndexError:
            self.album_cover = "default.jpg"

    def get (self):
        return {
            "source": self.source,
            "name": self.name,
            "artist": self.artist,
            "album": self.album,
            "album_cover": self.album_cover
        }

    def __str__(self):
        return str(self.get())

File: tesla_proj/java_ui/src/Spotify/Track.java
package Spotify;

import org.json.simple.JSONObject;

public class Track {
    private String source;
    private String name;
    private String artist;
    private String album;
    private String albumCover;

    public Track(JSONObject trackJson) {
        this.source = trackJson.get("source").toString();
        this.name = trackJson.get("name").toString();
        this.artist = trackJson.get("artist").toString();
        this.album = trackJson.get("album").toString();
        this.albumCover = trackJson.get("album_cover").toString();
    }

    @Override
    public String toString() {
        return "Name: " + this.name +
                "\nArtist: " + this.artist +
                "\nAlbum: " + this.album +
                "\nSource: " + this.source;
    }

    public String getSource() {
        return source;
    }

    public void setSource(String source) {
        this.source = source;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }

    public String getAlbum() {
        return album;
    }

    public void setAlbum(String album) {
        this.album = album;
    }

    public String getAlbumCover() {
        return this.albumCover;
    }

    public void setAlbumCover(String albumCover) {
        this.albumCover = albumCover;
    }
}
